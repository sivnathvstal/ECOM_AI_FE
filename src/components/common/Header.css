.header-title {
  flex-grow: 1;
  text-decoration: none;
  color: white;
}

.header-cart-button {
  margin-right: 16px;
}

.header-badge {
  right: -3px;
  top: 13px;
} 