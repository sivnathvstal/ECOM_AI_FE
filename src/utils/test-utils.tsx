import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider, createTheme } from '@material-ui/core';
import { QueryClient, QueryClientProvider } from 'react-query';
import { CartProvider } from '../context/CartContext';

// Create a theme for testing
const theme = createTheme({
  palette: {
    primary: {
      main: '#3f51b5',
    },
    secondary: {
      main: '#f50057',
    },
  },
});

// Create a client for React Query
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

// All providers wrapper
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>
        <CartProvider>
          <BrowserRouter>{children}</BrowserRouter>
        </CartProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// Re-export everything
export * from '@testing-library/react';

// Override render method
export { customRender as render }; 