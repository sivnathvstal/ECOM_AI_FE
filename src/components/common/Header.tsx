import React from 'react';
import { 
  AppBar, 
  Toolbar, 
  Typography, 
  IconButton, 
  Badge
} from '@material-ui/core';
import { ShoppingCart } from '@material-ui/icons';
import { Link as RouterLink } from 'react-router-dom';
import { useCart } from '../../context/CartContext';
import './Header.css';

const Header: React.FC = () => {
  const { totalItems } = useCart();

  return (
    <AppBar position="static">
      <Toolbar>
        <Typography
          component={RouterLink}
          to="/"
          variant="h6"
          className="header-title"
          data-testid="app-title"
        >
          E-Commerce Shop
        </Typography>
        <IconButton
          component={RouterLink}
          to="/cart"
          color="inherit"
          className="header-cart-button"
          aria-label="Shopping Cart"
          data-testid="cart-button"
        >
          <Badge badgeContent={totalItems} color="secondary" classes={{ badge: "header-badge" }}>
            <ShoppingCart />
          </Badge>
        </IconButton>
      </Toolbar>
    </AppBar>
  );
};

export default Header; 