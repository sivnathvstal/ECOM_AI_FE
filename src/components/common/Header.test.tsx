import React from 'react';
import { render, screen } from '../../utils/test-utils';
import Header from './Header';

describe('Header', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('displays the app title', () => {
    render(<Header />);
    expect(screen.getByTestId('app-title')).toHaveTextContent('E-Commerce Shop');
  });

  it('displays the cart button', () => {
    render(<Header />);
    const cartButton = screen.getByTestId('cart-button');
    expect(cartButton).toBeInTheDocument();
  });

  it('links to the correct routes', () => {
    render(<Header />);
    const titleLink = screen.getByTestId('app-title');
    const cartButton = screen.getByTestId('cart-button');

    expect(titleLink).toHaveAttribute('href', '/');
    expect(cartButton).toHaveAttribute('href', '/cart');
  });
}); 