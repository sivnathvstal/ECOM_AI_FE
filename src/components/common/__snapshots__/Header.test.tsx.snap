// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        class="MuiTypography-root header-title MuiTypography-h6"
        data-testid="app-title"
        href="/"
      >
        E-Commerce Shop
      </a>
      <a
        aria-disabled="false"
        aria-label="Shopping Cart"
        class="MuiButtonBase-root MuiIconButton-root header-cart-button MuiIconButton-colorInherit"
        data-testid="cart-button"
        href="/cart"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <span
              class="MuiBadge-badge header-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
            >
              0
            </span>
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </header>
</div>
`;
